circle.yml

```diff
@@ -9,12 +9,11 @@ dependencies:
   override:
     # Used by container_spec
     - docker pull alpine
-    - docker build -t=$PRIVATE_REGISTRY/$CIRCLE_PROJECT_REPONAME:b$CIRCLE_BUILD_NUM .
-    - bundle install
+    - make image
 
 test:
   override:
-    - bundle exec rake
+    - make test
```

Comment:
Thought of something else about why this kind of bugs me: I agree with Devon that having the desperate step in circle is nice. And I also agree having the task be dependent in the Makefile is important. I think doing *both* is a false advantage because, yes, you have the build in a separate step, but you are also polluting the test step output with the same build output all over again (except that every later will be cached), which is bad.

So on balance I still think we should either have separate circle steps *or* Makefile dependencies, but not both. Personally I lean towards keeping the Makefile dependencies since that seems like a more important use case for local development.